sympy/solvers/ode/single.py

```diff
@@ -135,6 +135,14 @@ def iter_numbered_constants(self, start=1, prefix='C') -> Iterator[Symbol]:
             atom_set |= {Symbol(str(f.func)) for f in func_set}
         return numbered_symbols(start=start, prefix=prefix, exclude=atom_set)
 
+    def is_autonomous(self):
```

Comment:
This could use `@cached_property`.

Reply:
Anything else? I don't want the tests to run again just because of a small change :)

Reply:
I left a couple more small comments but then it can be merged. Good work!

Reply:
Thank you for your time to review this!